src/StreamJsonRpc.Shared/Reflection/MethodNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have an alternate name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class MethodNameAttribute : Attribute
+    {
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="name">Alternate name of a method.</param>
```

Comment:
Can you describe what the significance of the alternate name is, both here and in the type's summary?
Specifically the details I was looking for but didn't see was whether this *replaces* the remote name of the method vs. adds another option.